BrowserKit/Sources/ToolbarKit/LocationView.swift

```diff
@@ -56,32 +78,103 @@ class LocationView: UIView, UITextFieldDelegate {
+
+        NSLayoutConstraint.activate(
+            [
+                gradientView.topAnchor.constraint(
+                    equalTo: urlTextField.topAnchor,
+                    constant: UX.gradientViewVerticalPadding
+                ),
+                gradientView.bottomAnchor.constraint(
+                    equalTo: urlTextField.bottomAnchor,
+                    constant: -UX.gradientViewVerticalPadding
+                ),
+                gradientView.leadingAnchor.constraint(equalTo: urlTextField.leadingAnchor),
+                gradientView.widthAnchor.constraint(equalToConstant: UX.gradientViewWidth),
+                gradientView.centerYAnchor.constraint(equalTo: urlTextField.centerYAnchor),
+
+                urlTextField.leadingAnchor.constraint(equalTo: leadingAnchor, constant: UX.horizontalSpace),
+                urlTextField.topAnchor.constraint(equalTo: topAnchor),
+                urlTextField.trailingAnchor.constraint(lessThanOrEqualTo: trailingAnchor, constant: -UX.horizontalSpace),
+                urlTextField.bottomAnchor.constraint(equalTo: bottomAnchor),
+            ]
+        )
+    }
+
+    private func setupGradientLayer() {
+        gradientLayer.locations = [0, 1]
+        gradientLayer.startPoint = CGPoint(x: 0.0, y: 0.5)
+        gradientLayer.endPoint = CGPoint(x: 1.0, y: 0.5)
+        gradientView.layer.addSublayer(gradientLayer)
+    }
+
+    private func updateGradientLayerFrame() {
+        let locationViewWidth = frame.width - (UX.horizontalSpace * 2)
+        let urlTextFieldWidth = urlTextField.frame.width
+        gradientLayer.frame = if urlTextFieldWidth >= locationViewWidth &&
+                                  !urlTextField.isFirstResponder { gradientView.bounds } else { CGRect() }
+    }
 
-        NSLayoutConstraint.activate([
-            urlTextField.leadingAnchor.constraint(equalTo: leadingAnchor, constant: UX.horizontalSpace),
-            urlTextField.topAnchor.constraint(equalTo: topAnchor),
-            urlTextField.trailingAnchor.constraint(lessThanOrEqualTo: trailingAnchor, constant: -UX.horizontalSpace),
-            urlTextField.bottomAnchor.constraint(equalTo: bottomAnchor),
-        ])
+    // MARK: - `urlTextField` Configuration
+    private func applyTruncationStyleToURLTextField() {
+        let paragraphStyle = NSMutableParagraphStyle()
+        paragraphStyle.lineBreakMode = .byTruncatingHead
+
+        let attributedString = NSMutableAttributedString(string: urlTextField.text ?? "")
+        attributedString.addAttribute(
+            .paragraphStyle,
+            value: paragraphStyle,
+            range: NSRange(
+                location: 0,
+                length: attributedString.length
+            )
+        )
+        urlTextField.attributedText = attributedString
     }
 
+    private func getHost(from stringURL: String?) -> String {
+        guard let stringURL,
+              let url = URL(string: stringURL) else { return "" }
```

Comment:
```suggestion
        guard let stringURL, let url = URL(string: stringURL) else { return "" }
```
Nit: this guard is short enough that we don't need two lines.